[src] Add accuracy authorization field to appUserTurnstile event

mr1sunshine opened the pull request.

frederoni commented on MapboxMobileEvents/Categories/CLLocationManager+MMEMobileEvents.m:
```diff
@@ -1,14 +1,25 @@
 #import "CLLocationManager+MMEMobileEvents.h"
-#import "MMEConstants.h"
+#import "../MMEConstants.h"
```
All of these headers are flattened and would end up in the same directory so dropping `../` must work.

mr1sunshine committed:
[tests] Fix Cedar test cases for MMELocationManagerSpec on iOS14

Starting from iOS14 we should call instance method "authorizatioStatus"
of CCLocationManager instead of static method. In order to pass test
cases we should mock accrodingly methods above.

mr1sunshine committed:
[src] Change locationAuthorization method to be compatible with iOS14

mr1sunshine committed:
[src] Add accuracy authorization field to appUserTurnsTile event

alfwatt reviewed (approved):
Looks good
